multiapps-controller-persistence/src/main/java/org/cloudfoundry/multiapps/controller/persistence/util/ClientKeyConfigurationHandler.java

```diff
@@ -0,0 +1,32 @@
+package org.cloudfoundry.multiapps.controller.persistence.util;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.text.MessageFormat;
+
+import org.bouncycastle.openssl.PEMParser;
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter;
+import org.cloudfoundry.multiapps.controller.persistence.Messages;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ClientKeyConfigurationHandler {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClientKeyConfigurationHandler.class);
+
+    public void createEncodedKeyFile(String pemKey, String fileName) {
+        try (PEMParser pemParser = new PEMParser(new StringReader(pemKey))) {
+            PEMToEncodedKeyConverter derKeyHandler = new PEMToEncodedKeyConverter();
```

Comment:
I am little bit confused. How `derKeyHandler` comes from? The object is named `PEMToEcnodedKeyConverter` so it is again converter from PEM to encoded format. Variable name should be different in my opinion.

Reply:
I agree that the variable should be renamed. The reason why it's named `derKeyHandler` is that the .der format is basically the binary version of a PEM extension, so an encoded pem file can be seen as a der file. But even then the name is confusing. Something along the lines of `pemKeyConverter` I think might be more appropriate